ttnn/cpp/ttnn/operations/moreh/moreh_nll_loss_backward/moreh_nll_loss_backward.hpp

```diff
@@ -0,0 +1,32 @@
+
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "ttnn/decorators.hpp"
+#include "ttnn/operations/core/compute_kernel/compute_kernel_config.hpp"
+
+namespace ttnn::operations::moreh::moreh_nll_loss_backward {
+
+struct MorehNllLossBackward {
+    static Tensor invoke(
+        const Tensor &target_tensor,
+        const Tensor &output_grad_tensor,
+        const bool reduction_mean,
+        const std::optional<const Tensor> weight_tensor,
+        const std::optional<const Tensor> input_grad_tensor,
+        const std::optional<const Tensor> divisor_tensor,
+        const std::optional<int32_t> ignore_index,
```

Comment:
https://github.com/pytorch/pytorch/blob/main/torch/nn/modules/loss.py#L243
According to the PyTorch code, this is not optional

The documentation specifies it as (int, optional), but the reason is not clear to me.

Reply:
I think is optional because if you don't pass the param, it use the default value, which is -100. Since ignore_index should be in the range of [0, C) this means that no index is ignore.

Reply:
```Python
    if ignore_index == None:
        nll_loss = torch.nn.NLLLoss(weight=torch_weight, reduction="none")
    else:
        nll_loss = torch.nn.NLLLoss(weight=torch_weight, ignore_index=ignore_index, reduction="none")
```
It seems that the `optional<int32_t>` type for ignore_index is causing unnecessary test code to be added in the code above.

Using the optional<int32_t> type would require additional code and tests for cases where ignore_index is None. Is there a reason to use this type?

```Python
import torch
import torch.nn.functional as F

# input is of size N x C = 3 x 5
input = torch.randn(3, 5, requires_grad=True)
# each element in target has to have 0 <= value < C
target = torch.tensor([1, 0, 4])
output = F.nll_loss(F.log_softmax(input, dim=1), target, ignore_index=None)

print(output)

TypeError: nll_loss_nd(): argument 'ignore_index' (position 5) must be int, not NoneType
```
I tried using None for ignore_index in PyTorch, and it seems that this feature is not supported.

Reply:
I have updated: 

In pytest: use -100 instead of None
<img width="415" alt="image" src="https://github.com/user-attachments/assets/fe1d5170-aef6-424e-851c-f6d3df6f7c42">

Pybind: 
<img width="320" alt="image" src="https://github.com/user-attachments/assets/bd9dd9c6-9c54-4dc5-9d2b-5cc3908e3c01">

C++ param: use int32_t instead of Optional<int32_t> then convert to uint32_t in operation_attributes_t as kernel runtime args required passsing uint32_t.
<img width="812" alt="image" src="https://github.com/user-attachments/assets/350db7c2-3073-4b6c-9d0a-45b296117189">